client/src/app/template-questionnaire/template-questionnaire.data.ts

```diff
@@ -1,27 +1,78 @@
+import * as answerKey from './answer-key.json';
+
+const answers = (answerKey as any).default;
+
 export interface QuestionnaireData {
   technologyType: string[];
   libraryPreparation: string[];
   identifyingOrganisms: string[];
   specimenType: string[];
+  experimentInfo: string;
+  preNatalQuantity: string;
+  donorsRelated: string;
+  specimenPurchased: string;
+  protocols: string[];
 }
 
-interface TypeSpec {
+export interface TypeSpec {
   schemaName: string;
-  excludeModules: string[];
-  excludeFields: string[];
+  includeModules: string[];
   embedProcess: boolean;
   linkSpec: {
     linkEntities: string[];
     linkProtocols: string[]
   };
 }
 
-
 export class TemplateSpecification {
-  types: TypeSpec[];
 
-  static convert(data: QuestionnaireData): TemplateSpecification {
-    return new TemplateSpecification();
+  private types = new Map<String, TypeSpec>();
+
+  static convert(questionnaire: QuestionnaireData): TemplateSpecification {
+    let specification = new TemplateSpecification();
```

Comment:
(small)
I would do `const _questionnaire = questionnaire as any;` here, and then use `_questionnaire` in the rest of the method. This will cast `questionnaire` into the `any` type, which allow it to be used like a dictionary/hashmap as below without the typescript compiler complaining. Alternatively, could write `QuestionnaireData` such that it implements Map<> and allows for expressions like `question in questionnaire` or `questionnaire[<some-string>]`

Ideally (mentioned in another PR in this repo), at some point we should consider using a reasonably strict tsconfig.json, otherwise we're essentially just writing type-unsafe javascript code.